# Titulo

## Explicacion

### Posibles comentarios

* **Importo libreria pandas**

In [15]:
#importando pandas
import pandas as pd 

* **Compruebo si leo bien el archivo de datos datosEEGTotal.csv**

In [16]:
pd.read_csv('../datos/datosEEGTotal.csv', sep=';')

Timestamp  Attention  Meditation   Delta  Theta  LowAlpha  HighAlpha  \
0        54127         84          48   25253  14225      3441       4172   
1        55056         78          60    4946   5021      8879      13002   
2        56038         66          75  102896  14838     12817       8565   
3        57043         61          87   57763  12541      6173       5120   
4        58203         37          84   21560  74162     11297       9217   
..         ...        ...         ...     ...    ...       ...        ...   
721     348567         81          37    4088  13782       756       5536   
722     355018         84          20    3490  13947      2299       2051   
723     358034         84          38    3619   8525      5623       6665   
724     361108         83          51  116972   5464      1405       6395   
725     362635         57          44  120133  59585      2946       1584   

     LowBeta  HighBeta  LowGamma  HighGamma  Signal      Key  
0       1881      9630      6863       5745       0  LButton  
1        879      6367      4314       3150       0      NaN  
2       2890     14525      9144      23072       0      NaN  
3       3238     10192      9048       6663       0      NaN  
4       4844      1945      2644       1678       0      NaN  
..       ...       ...       ...        ...     ...      ...  
721     7114     24230     12942      14061       0    Right  
722     3315     22789     17542       9966       0    Right  
723     5345     31352     22837      12553       0    Right  
724     2285      7186      5351       1219       0    Right  
725     5438      6210      3140       1884       0      NaN  

[726 rows x 13 columns]

* **Guardo los datos en datosEEGTotal_csv**

In [17]:
datosEEGTotal_csv = pd.read_csv('../datos/datosEEGTotal.csv', sep=';')

* **Compruebo los datos en el dataset datosEEGTotal_csv**

In [18]:
datosEEGTotal_csv

Timestamp  Attention  Meditation   Delta  Theta  LowAlpha  HighAlpha  \
0        54127         84          48   25253  14225      3441       4172   
1        55056         78          60    4946   5021      8879      13002   
2        56038         66          75  102896  14838     12817       8565   
3        57043         61          87   57763  12541      6173       5120   
4        58203         37          84   21560  74162     11297       9217   
..         ...        ...         ...     ...    ...       ...        ...   
721     348567         81          37    4088  13782       756       5536   
722     355018         84          20    3490  13947      2299       2051   
723     358034         84          38    3619   8525      5623       6665   
724     361108         83          51  116972   5464      1405       6395   
725     362635         57          44  120133  59585      2946       1584   

     LowBeta  HighBeta  LowGamma  HighGamma  Signal      Key  
0       1881      9630      6863       5745       0  LButton  
1        879      6367      4314       3150       0      NaN  
2       2890     14525      9144      23072       0      NaN  
3       3238     10192      9048       6663       0      NaN  
4       4844      1945      2644       1678       0      NaN  
..       ...       ...       ...        ...     ...      ...  
721     7114     24230     12942      14061       0    Right  
722     3315     22789     17542       9966       0    Right  
723     5345     31352     22837      12553       0    Right  
724     2285      7186      5351       1219       0    Right  
725     5438      6210      3140       1884       0      NaN  

[726 rows x 13 columns]

* **Examino cuantos datos estan perdidos en el dataset datosEEGTotal_csv**

In [28]:
datosEEGTotal_csv.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 726 entries, 0 to 725
Data columns (total 13 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Timestamp   726 non-null    int64 
 1   Attention   726 non-null    int64 
 2   Meditation  726 non-null    int64 
 3   Delta       726 non-null    int64 
 4   Theta       726 non-null    int64 
 5   LowAlpha    726 non-null    int64 
 6   HighAlpha   726 non-null    int64 
 7   LowBeta     726 non-null    int64 
 8   HighBeta    726 non-null    int64 
 9   LowGamma    726 non-null    int64 
 10  HighGamma   726 non-null    int64 
 11  Signal      726 non-null    int64 
 12  Key         513 non-null    object
dtypes: int64(12), object(1)
memory usage: 73.9+ KB


Para los 726 registros, indica para cada columna los datos que no son nulos para la columna Key existen 513 no nulos, por lo tanto, hay 213 nulos

* **Compruebo los datos missing en la columna Key**

In [50]:
datosEEGTotal_csv[pd.isnull(datosEEGTotal_csv.Key)]

Timestamp  Attention  Meditation   Delta  Theta  LowAlpha  HighAlpha  \
1        55056         78          60    4946   5021      8879      13002   
2        56038         66          75  102896  14838     12817       8565   
3        57043         61          87   57763  12541      6173       5120   
4        58203         37          84   21560  74162     11297       9217   
5        59952         48          63    2540   1384       692        467   
..         ...        ...         ...     ...    ...       ...        ...   
717     341968         97          51    1649   4652      1007       4496   
718     342914         96          53    2355   3428      2653       2048   
719     343917         93          51    7732   9661      4148       3043   
720     345017         93          30    6033  11388      1086       1898   
725     362635         57          44  120133  59585      2946       1584   

     LowBeta  HighBeta  LowGamma  HighGamma  Signal  Key  
1        879      6367      4314       3150       0  NaN  
2       2890     14525      9144      23072       0  NaN  
3       3238     10192      9048       6663       0  NaN  
4       4844      1945      2644       1678       0  NaN  
5       1313      2271      2560        445       0  NaN  
..       ...       ...       ...        ...     ...  ...  
717     1952     51386     46310      19608       0  NaN  
718     3748     23375     28816      10428       0  NaN  
719     1962     29014     50664      24171       0  NaN  
720     3777     22194     34869       4480       0  NaN  
725     5438      6210      3140       1884       0  NaN  

[213 rows x 13 columns]

* **Reemplazo los datos missing en la columna Key y lo guardo en otro dataset**

In [79]:
datosEEGTotal_clean = datosEEGTotal_csv.fillna("None")

* **Examino cuantos datos estan perdidos en el dataset datosEEGTotalRelleno**

In [80]:
datosEEGTotal_clean.info(verbose=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 726 entries, 0 to 725
Data columns (total 13 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Timestamp   726 non-null    int64 
 1   Attention   726 non-null    int64 
 2   Meditation  726 non-null    int64 
 3   Delta       726 non-null    int64 
 4   Theta       726 non-null    int64 
 5   LowAlpha    726 non-null    int64 
 6   HighAlpha   726 non-null    int64 
 7   LowBeta     726 non-null    int64 
 8   HighBeta    726 non-null    int64 
 9   LowGamma    726 non-null    int64 
 10  HighGamma   726 non-null    int64 
 11  Signal      726 non-null    int64 
 12  Key         726 non-null    object
dtypes: int64(12), object(1)
memory usage: 73.9+ KB


No hay ningun dato perdido en el nuevo dataset

* **Examino si hay datos con diferentes nombres para unificarlos en el dataset datosEEGTotal_clean**

In [92]:
datosEEGTotal_clean.groupby('Key').Key.count()

Key
Down       118
LButton      6
Left       126
None       213
Right      117
Up         146
Name: Key, dtype: int64

LButton y Left se podrian unificar en Left para seguir la nomenclatura de las otras agrupaciones

* **Modifico en la columna Key los nombres que sean LButton por Left y creo nuevo dataset datosEEGTotal**

In [87]:
datosEEGTotal = datosEEGTotal_clean.replace({"LButton": 'Left'})

* **Examino si han quedado datos con diferentes nombres para unificarlos en el nuevo dataset datosEEGTotal**

In [89]:
datosEEGTotal.groupby('Key').Key.count()

Key
Down     118
Left     132
None     213
Right    117
Up       146
Name: Key, dtype: int64

In [91]:
datosEEGTotal.Key.unique() #Tambien se podria revisar con unique()

array(['Left', 'None', 'Up', 'Down', 'Right'], dtype=object)

In [93]:
datosEEGTotal.head()

Timestamp  Attention  Meditation   Delta  Theta  LowAlpha  HighAlpha  \
0      54127         84          48   25253  14225      3441       4172   
1      55056         78          60    4946   5021      8879      13002   
2      56038         66          75  102896  14838     12817       8565   
3      57043         61          87   57763  12541      6173       5120   
4      58203         37          84   21560  74162     11297       9217   

   LowBeta  HighBeta  LowGamma  HighGamma  Signal   Key  
0     1881      9630      6863       5745       0  Left  
1      879      6367      4314       3150       0  None  
2     2890     14525      9144      23072       0  None  
3     3238     10192      9048       6663       0  None  
4     4844      1945      2644       1678       0  None